### DISCUSSION TITLE
Best practice for removing type errors for references like auth, user etc.

### DISCUSSION THREAD

**Discussion by sugoidesune** [vuejs/vuefire#1539] [Questions and Help] [answered]
![image](https://github.com/vuejs/vuefire/assets/20317691/c0995a59-31d9-406d-8f41-9376b87576c9)

I know some workarounds but really curious how everyone else deals with this in a clean way.

**Comment by ralacerda** [comment, marked as answer]
This happens because the Auth module is not avaliable on the server, so it can be null.
When you call `useFirebaseAuth` you can include the[ Non-null assertion operator](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#non-null-assertion-operator) (`!`)

```ts
const auth = useFirebaseAuth()!
```

[The documentation includes an tip about this](https://vuefire.vuejs.org/guide/auth.html#Auth-instance)
